dspace-server-webapp/src/main/java/org/dspace/app/rest/OidcRestController.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest;
+
+import java.io.IOException;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.model.AuthnRest;
+import org.dspace.services.ConfigurationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Link;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Rest controller that handles redirect after OIDC authentication succeded.
+ *
+ * @author Luca Giamminonni (luca.giamminonni at 4science.it)
+ */
+@RestController
+@RequestMapping(value = "/api/" + AuthnRest.CATEGORY + "/oidc")
+public class OidcRestController {
```

Comment:
I'll just note here that the strategy of having both an `OidcRestController` **and** and `OidcAuthenticationFilter` might result in this "two auth cookies" bug that we hit with Shibboleth recently: https://github.com/DSpace/DSpace/pull/8032   Essentially, with Shibboleth, the controller & filter were both somehow creating an Auth Cookie, and fix was to remove the Controller entirely, and move all the logic into the Filter (see that PR as an example). 

While I cannot test OIDC to see if the same behavior will occur, I wanted to note here that this might be an issue we'll have to fix, as the OIDC code looks very similar to our old Shibboleth code where that error occurred.

Reply:
If this is still a concern, I can try to tackle the change. I'll see if I can translate the changes in #8032 to this plugin